excellent/legacy.go

```diff
@@ -107,6 +121,39 @@ func (v *varMapper) String() string {
 	return v.base
 }
 
+// Migration of @extra requires its own mapper because it can map differently depending on the containing flow
+type extraMapper struct {
+	varMapper
+
+	path    string
+	extraAs ExtraVarsMapping
+}
+
+func (m *extraMapper) Resolve(key string) interface{} {
+	newPath := []string{}
+	if m.path != "" {
+		newPath = append(newPath, m.path)
+	}
+	newPath = append(newPath, key)
+	return &extraMapper{extraAs: m.extraAs, path: strings.Join(newPath, ".")}
+}
+
+func (m *extraMapper) Default() interface{} {
+	return m
+}
+
+func (m *extraMapper) String() string {
+	switch m.extraAs {
+	case ExtraAsWebhookJSON:
+		return fmt.Sprintf("run.webhook.json.%s", m.path)
+	case ExtraAsTriggerParams:
+		return fmt.Sprintf("trigger.params.%s", m.path)
+	case ExtraAsFunction:
```

Comment:
I added three different ways to map `@extra` because although the `@(if(...))` solution should work in all cases, it seemed nice to map to a simple expression if the flow didn't have any webhook actions.

But making that work with legacy flow parsing is gonna require a significant work of legacy flow parsing (e.g. UnmarshalJSON methods with no access to state about the wider flow), so I wanna check how people feel about this first.